Sources/SwiftWin32/CG/Rect.swift

```diff
@@ -22,9 +22,29 @@ public struct Rect {
               size: Size(width: Double(width), height: Double(height)))
   }
 
-  func applying(_ transform: AffineTransform) -> Rect {
-    return Rect(origin: self.origin.applying(transform),
-                size: self.size.applying(transform))
+  public func applying(_ transform: AffineTransform) -> Rect {
+    let points: [Point] = [
+      (self.origin + Point(x: self.size.width, y: 0)).applying(transform),
+      (self.origin + Point(x: 0, y: self.size.height)).applying(transform),
+      (self.origin + Point(x: self.size.width, y: self.size.height)).applying(transform),
+    ]
+
+    let transformedOrigin = self.origin.applying(transform)
+
+    var minX = transformedOrigin.x
+    var minY = transformedOrigin.y
+    var maxX = transformedOrigin.x
+    var maxY = transformedOrigin.y
+
+    points.forEach { point in
+      minX = min(minX, point.x)
+      maxX = max(maxX, point.x)
+      minY = min(minY, point.y)
+      maxY = max(maxY, point.y)
+    }
+
+    return Rect(origin: Point(x: minX, y: minY),
+                size: Size(width: maxX - minX, height: maxY - minY))
```

Comment:
Sorry, now that there is much less noise, this seems far easier to follow what it is trying to do.  I think that this can be even more concise:

```suggestion
    let points: [Point] = [
      self.origin,
      self.origin + Point(x: self.size.width, y: 0),
      self.origin + Point(x: 0, y: self.size.height),
      self.origin + Point(x: self.size.width, y: self.size.height),
    ].map { $0.applying(transform) }

    let xs: [Double] = points.map { $0.x }
    let ys: [Double] = points.map { $0.y }
    
    return Rect(origin: Point(x: xs.min(), y: ys.min()),
                size: Size(width: xs.max() - xs.min(), height: ys.max() - ys.min())
```

I want to think a little bit more about this, because this seems like it could be even simpler.  But, this definitely is far easier to follow IMO: you have the four points defining the rectangle, which you transform given a transformation.  You then create the rectangle from the minimum x, minimum y, and and the difference of the maximum and minimum xs and ys.  Given that there are 4 points, known statically, this seems like this should be something that should be possible to implement without branching.